internal/zinc-compile-core/src/main/scala/sbt/internal/inc/javac/LocalJava.scala

```diff
@@ -274,6 +269,17 @@ final class WriteReportingFileManager(
     val output = super.getJavaFileForOutput(location, className, kind, sibling)
     new WriteReportingJavaFileObject(output, classFileManager)
   }
+
+  // Fixes #185, #684
+  override def isSameFile(a: FileObject, b: FileObject): Boolean = {
+    def unwrap(fo: FileObject): FileObject = {
+      fo match {
```

Comment:
Maybe add a case here where `fo` is `null`?

Reply:
Ok, I'll do. Good catch.

Reply:
Looked at this for the second time, and it is not needed.
If it is `null`, it will unwrap to `null` by the second case.

Reply:
Could you add 

```scala
Test / testOptions += Tests.Argument(TestFrameworks.ScalaCheck, "-verbosity", "2"),
```

under https://github.com/sbt/zinc/blob/7673b5e7642c8e1edd022f164213861e15c85e85/build.sbt#L67 to print the stack trace then?